.github/workflows/build_and_test_dbt.yaml

```diff
@@ -0,0 +1,122 @@
+name: Build and test dbt
+
+on:
+  pull_request:
+    branches: [master, data-catalog]
+    # Specifying event types manually allows us to run this flow when the
+    # PR is closed so that we can clean up staging dbt resources
+    types:
+      - opened
+      - synchronize
+      - closed
+      - reopened
+  push:
+    branches: [master, data-catalog]
+
+jobs:
+  build-and-test-dbt:
+    runs-on: ubuntu-latest
+    # These permissions are needed to interact with GitHub's OIDC Token endpoint
+    # so that we can authenticate with AWS
+    permissions:
+      id-token: write
+      contents: read
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Install dbt requirements
+        uses: ./.github/actions/install_dbt_requirements
+
+      - name: Load environment variables
+        uses: ./.github/actions/load_environment_variables
+
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v2
+        with:
+          role-to-assume: ${{ secrets.AWS_IAM_ROLE_TO_ASSUME_ARN }}
+          aws-region: us-east-1
+
+      - name: Set environment for branch
+        run: |
+          if [[ $GITHUB_REF_NAME == 'master' ]]; then
+            echo "On master branch"
+            {
+              echo "TARGET=prod";
+              echo "CACHE_KEY=master";
+            } >> "$GITHUB_ENV"
+          elif [[ $GITHUB_REF_NAME == 'data-catalog' ]]; then
+            echo "On data catalog branch"
+            {
+              echo "TARGET=ci";
+              echo "CACHE_KEY=data-catalog";
+              echo "GITHUB_HEAD_REF=data-catalog";
+            } >> "$GITHUB_ENV"
+          else
+            echo "On pull request branch"
+            {
+              echo "TARGET=ci";
+              echo "CACHE_KEY=$GITHUB_HEAD_REF";
+            } >> "$GITHUB_ENV"
+          fi
+        shell: bash
+
+      - name: Cache dbt manifest
+        id: cache
+        uses: actions/cache@v3
+        with:
+          path: ${{ env.MANIFEST_DIR }}
+          key: ${{ env.CACHE_NAME }}-${{ env.CACHE_KEY }}
+          restore-keys: |
+            ${{ env.CACHE_NAME }}-data-catalog
+            ${{ env.CACHE_NAME }}-master
```

Comment:
The `restore-keys` setting here means that we will fall back to the `data-catalog` or `master` branch caches if the PR branch cache does not exist (which we only ever expect to happen on the first pushes to a PR, before the PR has had a successful run).